### ISSUE TITLE
Remote Container Extensions are Not Installed

### ISSUE THREAD

**Issue by sbryfcz** [microsoft/vscode-remote-release#4268]
Maybe this is a problem with VS Code and not this repo, but when I start this container and connect to it, the extension for python is not being installed. Not sure if you experience similar issues.

- VSCode Version: 1.52.1
- Local OS Version: Darwin x64 19.6.0
- Remote OS Version: Alpine I believe. From: https://github.com/microsoft/vscode-remote-try-python/blob/master/.devcontainer/Dockerfile
- Remote Extension/Connection Type: Docker

Steps to Reproduce:
1. Clone the vscode-remote-try-python repo
2. Open VS Code
3. Open the repo as remote container
4. Wait for remote connection to the container
5. Check which remote extensions are active and notice that none of the extensions for the devcontainer.json extensions are present in the remote container

Does this issue occur when you try this locally?: Yes? Not sure how to test
Does this issue occur when you try this locally and all extensions are disabled?: Yes? Extensions are what I'm trying to test

It seems that the container starts up fine, but none of the extensions are installing. I can use the VS Code GUI to install but if I tear the image down and up again, they aren't present. Obviously, this slows down development a good deal. The console doesn't seem to give errors that seem actionable, but maybe I'm missing something. See below.

Open to suggestions.

```[3 ms] Remote-Containers 0.154.1 in VS Code 1.52.1 (ea3859d4ba2f3e577a159bc91e3074c5d85c0523).
[20 ms] Start: Resolving Remote
[21 ms] Setting up container for folder or workspace: /Users/sam.bryfczynski/Documents/Projects/vscode-remote-try-python

[23 ms] Start: Check Docker is running
[23 ms] Start: Run: docker version --format {{.Server.APIVersion}}
[156 ms] Server API version: 1.41
[170 ms] Start: Run: git rev-parse --show-cdup
[181 ms] Start: Run: docker ps -q -a --filter label=vsch.local.folder=/Users/sam.bryfczynski/Documents/Projects/vscode-remote-try-python --filter label=vsch.quality=stable
[359 ms] Start: Run: docker build -f /Users/sam.bryfczynski/Documents/Projects/vscode-remote-try-python/.devcontainer/Dockerfile -t vsc-vscode-remote-try-python-debdd202e60137db0a3e10cc89a52855 --build-arg VARIANT=3 /Users/sam.bryfczynski/Documents/Projects/vscode-remote-try-python
Sending build context to Docker daemon  121.3kB
Step 1/6 : ARG VARIANT="3"
Step 2/6 : FROM mcr.microsoft.com/vscode/devcontainers/python:0-${VARIANT}
 ---> 3ed7ccb0d202
Step 3/6 : ENV PIP_TARGET=/usr/local/pip-global
 ---> Using cache
 ---> 2bfe922928c1
Step 4/6 : ENV PYTHONPATH=${PIP_TARGET}:${PYTHONPATH}
 ---> Using cache
 ---> 653a8cf3f70f
Step 5/6 : ENV PATH=${PIP_TARGET}/bin:${PATH}
 ---> Using cache
 ---> 00f3fdb45260
Step 6/6 : RUN if ! cat /etc/group | grep -e "^pip-global:" > /dev/null 2>&1; then groupadd -r pip-global; fi     && usermod -a -G pip-global vscode     && umask 0002 && mkdir -p ${PIP_TARGET}     && chown :pip-global ${PIP_TARGET}     && ( [ ! -f "/etc/profile.d/00-restore-env.sh" ] || sed -i -e "s/export PATH=/export PATH=\/usr\/local\/pip-global:/" /etc/profile.d/00-restore-env.sh )
 ---> Using cache
 ---> cddda8a10f1e
Successfully built cddda8a10f1e
Successfully tagged vsc-vscode-remote-try-python-debdd202e60137db0a3e10cc89a52855:latest
[681 ms] Start: Run: docker events --format {{json .}} --filter status=start
[684 ms] Start: Starting container
[685 ms] Start: Run: docker run -a STDOUT -a STDERR --mount type=bind,source=/Users/sam.bryfczynski/Documents/Projects/vscode-remote-try-python,target=/workspaces/vscode-remote-try-python,consistency=cached --mount type=volume,src=vscode,dst=/vscode -l vsch.quality=stable -l vsch.remote.devPort=0 -l vsch.local.folder=/Users/sam.bryfczynski/Documents/Projects/vscode-remote-try-python --entrypoint /bin/sh vsc-vscode-remote-try-python-debdd202e60137db0a3e10cc89a52855 -c echo Container started ; trap "exit 0" 15; while sleep 1 & wait $!; do :; done
[1485 ms] Start: Run: docker ps -q -a --filter label=vsch.local.folder=/Users/sam.bryfczynski/Documents/Projects/vscode-remote-try-python --filter label=vsch.quality=stable
Container started
[1672 ms] Start: Run: docker inspect --type container 65a650481ec4
[1819 ms] Start: Inspecting container
[1819 ms] Start: Run: docker inspect --type container 65a650481ec4fd2ea0f2e6f09af3b32b41c3706d4d14f4057dfa02b838edb070
[1988 ms] Start: Run: docker exec -i -u vscode -e VSCODE_REMOTE_CONTAINERS_SESSION=55deb90c-99cf-4d79-bfd7-c07bb7f457b01609786611714 65a650481ec4fd2ea0f2e6f09af3b32b41c3706d4d14f4057dfa02b838edb070 /bin/sh
[1992 ms] Start: Run in container: uname -m
[2409 ms] x86_64
[2409 ms] 
[2409 ms] Start: Run in container: (cat /etc/os-release || cat /usr/lib/os-release) 2>/dev/null
[2427 ms] PRETTY_NAME="Debian GNU/Linux 10 (buster)"
NAME="Debian GNU/Linux"
VERSION_ID="10"
VERSION="10 (buster)"
VERSION_CODENAME=buster
ID=debian
HOME_URL="https://www.debian.org/"
SUPPORT_URL="https://www.debian.org/support"
BUG_REPORT_URL="https://bugs.debian.org/"
[2428 ms] 
[2428 ms] Start: Run in container: cat /etc/passwd
[2437 ms] Start: Updating configuration state
[2458 ms] Start: Setup shutdown monitor
[2465 ms] Forking shutdown monitor: /Users/sam.bryfczynski/.vscode/extensions/ms-vscode-remote.remote-containers-0.154.1/dist/shutdownMonitorProcess /var/folders/t8/6kzxhfnj1mz1fghzftlssx8n90tbzp/T/vscode-remote-containers-8651222873fefd3ec2a15b2b5c7f16da58443440.sock singleContainer Info /Users/sam.bryfczynski/Library/Application Support/Code/logs/20210104T135616/exthost1/ms-vscode-remote.remote-containers 1609786612439
[2470 ms] Start: Run in container: test -d /home/vscode/.vscode-server
[2496 ms] 
[2497 ms] 
[2497 ms] Exit code 1
[2497 ms] Start: Run in container: test -d /home/vscode/.vscode-remote
[2500 ms] 
[2500 ms] 
[2500 ms] Exit code 1
[2500 ms] Start: Run in container: set -o noclobber ; mkdir -p '/home/vscode/.vscode-server/data/Machine' && { > '/home/vscode/.vscode-server/data/Machine/.writeMachineSettingsMarker' ; } 2> /dev/null
[2523 ms] 
[2523 ms] 
[2523 ms] Start: Run in container: mkdir -p '/home/vscode/.vscode-server/data/Machine' && cat >'/home/vscode/.vscode-server/data/Machine/settings.json' <<'settingsJSON'
[2533 ms] 
[2533 ms] 
[2534 ms] Start: Run in container: test -d /home/vscode/.vscode-server/bin/ea3859d4ba2f3e577a159bc91e3074c5d85c0523
[2538 ms] 
[2538 ms] 
[2538 ms] Exit code 1
[2539 ms] Start: Run in container: test -d /vscode/vscode-server/bin/x64/ea3859d4ba2f3e577a159bc91e3074c5d85c0523
[2543 ms] 
[2543 ms] 
[2543 ms] Exit code 1
[2544 ms] Installing VS Code Server for commit ea3859d4ba2f3e577a159bc91e3074c5d85c0523
[2544 ms] Start: Run: docker exec -i -u root 65a650481ec4fd2ea0f2e6f09af3b32b41c3706d4d14f4057dfa02b838edb070 /bin/sh
[2554 ms] Start: Run in container: mkdir -p /vscode/vscode-server/bin/x64/ea3859d4ba2f3e577a159bc91e3074c5d85c0523_1609786614983
[2951 ms] 
[2952 ms] 
[2953 ms] Start: Installing VS Code Server
[3010 ms] Start: Run in container: (dd iflag=fullblock bs=8192 count=3723 2>/dev/null; dd iflag=fullblock bs=895 count=1 2>/dev/null) | tar --no-same-owner -xz --strip-components 1 -C /vscode/vscode-server/bin/x64/ea3859d4ba2f3e577a159bc91e3074c5d85c0523_1609786614983
[4504 ms] 
[4504 ms] 
[4504 ms] Start: Run in container: mv -n /vscode/vscode-server/bin/x64/ea3859d4ba2f3e577a159bc91e3074c5d85c0523_1609786614983 /vscode/vscode-server/bin/x64/ea3859d4ba2f3e577a159bc91e3074c5d85c0523
[4510 ms] 
[4510 ms] 
[4510 ms] Start: Run in container: mkdir -p '/home/vscode/.vscode-server/bin' && ln -s '/vscode/vscode-server/bin/x64/ea3859d4ba2f3e577a159bc91e3074c5d85c0523' '/home/vscode/.vscode-server/bin/ea3859d4ba2f3e577a159bc91e3074c5d85c0523'
[4518 ms] 
[4518 ms] 
[4518 ms] Start: Run in container: cd '/vscode/vscode-server/bin/x64' && ls -t | tail -n +10 | xargs rm -rf
[4518 ms] Start: Launching Remote-Containers helper.
[4518 ms] Start: Run: gpgconf --list-dir agent-extra-socket
[4525 ms] Start: Run in container: # Test for /home/vscode/.ssh/known_hosts and ssh
[4530 ms] 
[4530 ms] 
[4530 ms] Start: Run in container: cat <<'EOF-/tmp/vscode-remote-containers-8e5415192432cd6be339ed81b76eec269ad2fa93.js' >/tmp/vscode-remote-containers-8e5415192432cd6be339ed81b76eec269ad2fa93.js
[4530 ms] 
[4530 ms] 
[4545 ms] 
[4545 ms] 
[4545 ms] Start: Run in container: # Copy /Users/sam.bryfczynski/.ssh/known_hosts to /home/vscode/.ssh/known_hosts
[4551 ms] 
[4551 ms] 
[4551 ms] Start: Run in container: cat <<'EOF-/tmp/vscode-remote-containers-server-8e5415192432cd6be339ed81b76eec269ad2fa93.js' >/tmp/vscode-remote-containers-server-8e5415192432cd6be339ed81b76eec269ad2fa93.js
[4551 ms] Start: Run in container: command -v git >/dev/null 2>&1 && git config --system credential.helper '!f() { /home/vscode/.vscode-server/bin/ea3859d4ba2f3e577a159bc91e3074c5d85c0523/node /tmp/vscode-remote-containers-8e5415192432cd6be339ed81b76eec269ad2fa93.js $*; }; f' || true
[4569 ms] 
[4570 ms] 
[4570 ms] Start: Run: docker exec -i -u vscode -e REMOTE_CONTAINERS_SOCKETS=["/tmp/vscode-ssh-auth-8e5415192432cd6be339ed81b76eec269ad2fa93.sock"] -e REMOTE_CONTAINERS_IPC=/tmp/vscode-remote-containers-ipc-8e5415192432cd6be339ed81b76eec269ad2fa93.sock 65a650481ec4fd2ea0f2e6f09af3b32b41c3706d4d14f4057dfa02b838edb070 /home/vscode/.vscode-server/bin/ea3859d4ba2f3e577a159bc91e3074c5d85c0523/node /tmp/vscode-remote-containers-server-8e5415192432cd6be339ed81b76eec269ad2fa93.js
[4605 ms] 
[4606 ms] 
[4606 ms] Start: Run in container: set -o noclobber ; mkdir -p '/home/vscode/.vscode-server/data/Machine' && { > '/home/vscode/.vscode-server/data/Machine/.postCreateCommandMarker' ; } 2> /dev/null
[4619 ms] 
[4620 ms] 
[4620 ms] Start: Run in container: test -f /home/vscode/.vscode-server/.postCreateCommandMarker
[4623 ms] 
[4623 ms] 
[4623 ms] Exit code 1
[5018 ms] Start: Run: docker exec -i -u vscode -e SSH_AUTH_SOCK=/tmp/vscode-ssh-auth-8e5415192432cd6be339ed81b76eec269ad2fa93.sock -e REMOTE_CONTAINERS_IPC=/tmp/vscode-remote-containers-ipc-8e5415192432cd6be339ed81b76eec269ad2fa93.sock -e REMOTE_CONTAINERS=true -w /workspaces/vscode-remote-try-python 65a650481ec4fd2ea0f2e6f09af3b32b41c3706d4d14f4057dfa02b838edb070 /bin/sh -c pip3 install -r requirements.txt
[6661 ms] Collecting flask
[6879 ms]   Downloading Flask-1.1.2-py2.py3-none-any.whl (94 kB)
[7317 ms] Collecting click>=5.1
[7354 ms]   Downloading click-7.1.2-py2.py3-none-any.whl (82 kB)
[7376 ms] Collecting itsdangerous>=0.24
[7411 ms]   Downloading itsdangerous-1.1.0-py2.py3-none-any.whl (16 kB)
[7421 ms] Collecting Jinja2>=2.10.1
[7461 ms]   Downloading Jinja2-2.11.2-py2.py3-none-any.whl (125 kB)
[7588 ms] Collecting MarkupSafe>=0.23
[7621 ms]   Downloading MarkupSafe-1.1.1.tar.gz (19 kB)
[7982 ms] Collecting Werkzeug>=0.15
[8021 ms]   Downloading Werkzeug-1.0.1-py2.py3-none-any.whl (298 kB)
[8115 ms] Building wheels for collected packages: MarkupSafe
[8116 ms]   Building wheel for MarkupSafe (setup.py): started
[9123 ms]   Building wheel for MarkupSafe (setup.py): finished with status 'done'
[9124 ms]   Created wheel for MarkupSafe: filename=MarkupSafe-1.1.1-cp39-cp39-linux_x86_64.whl size=32244 sha256=4b7d8239bb3a2690b82de8df482a43afe3b877420bdbb182dd64565eba357126
  Stored in directory: /home/vscode/.cache/pip/wheels/e0/19/6f/6ba857621f50dc08e084312746ed3ebc14211ba30037d5e44e
[9125 ms] Successfully built MarkupSafe
[9201 ms] Installing collected packages: MarkupSafe, Werkzeug, Jinja2, itsdangerous, click, flask
[9568 ms] Successfully installed Jinja2-2.11.2 MarkupSafe-1.1.1 Werkzeug-1.0.1 click-7.1.2 flask-1.1.2 itsdangerous-1.1.0
[9993 ms] Start: Run in container: # Test for /home/vscode/.gitconfig and git
[9998 ms] 
[9998 ms] 
[9998 ms] Start: Run in container: # Copy /Users/sam.bryfczynski/.gitconfig to /home/vscode/.gitconfig
[10006 ms] 
[10006 ms] 
[10006 ms] Start: Run in container: command -v git >/dev/null 2>&1 && git config --global credential.helper '!f() { /home/vscode/.vscode-server/bin/ea3859d4ba2f3e577a159bc91e3074c5d85c0523/node /tmp/vscode-remote-containers-8e5415192432cd6be339ed81b76eec269ad2fa93.js $*; }; f' || true
[10020 ms] 
[10020 ms] 
[10021 ms] Start: Preparing Extensions
[10021 ms] Start: Run in container: set -o noclobber ; mkdir -p '/home/vscode/.vscode-server/data/Machine' && { > '/home/vscode/.vscode-server/data/Machine/.installExtensionsMarker' ; } 2> /dev/null
[10027 ms] 
[10027 ms] 
[10028 ms] Extensions cache, install extensions: ms-python.python
[10028 ms] Start: Run in container: test -d /home/vscode/.vscode-server/extensionsCache && ls /home/vscode/.vscode-server/extensionsCache || true
[10032 ms] 
[10032 ms] 
[10032 ms] Start: Run in container: test -d /vscode/vscode-server/extensionsCache && ls /vscode/vscode-server/extensionsCache || true
[10034 ms] 
[10034 ms] 
[10035 ms] Start: Run in container: for pid in `cd /proc && ls -d [0-9]*`; do { echo $pid ; readlink /proc/$pid/cwd ; readlink /proc/$pid/ns/mnt ; cat /proc/$pid/stat | tr "
[10144 ms] Start: Starting VS Code Server
[10144 ms] Start: Run in container: /home/vscode/.vscode-server/bin/ea3859d4ba2f3e577a159bc91e3074c5d85c0523/server.sh --force-disable-user-env --use-host-proxy --port 0 --extensions-download-dir /home/vscode/.vscode-server/extensionsCache --install-extension ms-python.python --start-server
[10218 ms] 

*
* Visual Studio Code Server
*
* Reminder: You may only use this software with Visual Studio family products,
* as described in the license https://aka.ms/vscode-remote/license
*

Extension host agent listening on 44191

[10218 ms] Start: Run in container: echo 44191 >/home/vscode/.vscode-server/data/Machine/.devport
[10222 ms] 
[10222 ms] 
[10223 ms] Forwarding local port 44191 to container port 44191
[10269 ms] Start: Run: docker exec -i -u vscode -e VSCODE_REMOTE_CONTAINERS_SESSION=55deb90c-99cf-4d79-bfd7-c07bb7f457b01609786611714 65a650481ec4fd2ea0f2e6f09af3b32b41c3706d4d14f4057dfa02b838edb070 /home/vscode/.vscode-server/bin/ea3859d4ba2f3e577a159bc91e3074c5d85c0523/node -e 
[10282 ms] Start: Run: docker exec -i -u vscode -e VSCODE_REMOTE_CONTAINERS_SESSION=55deb90c-99cf-4d79-bfd7-c07bb7f457b01609786611714 65a650481ec4fd2ea0f2e6f09af3b32b41c3706d4d14f4057dfa02b838edb070 /home/vscode/.vscode-server/bin/ea3859d4ba2f3e577a159bc91e3074c5d85c0523/node -e 
[10629 ms] [18:57:03] Extension host agent started.
[10642 ms] Installing extensions...
[10993 ms] [18:57:03] [::ffff:127.0.0.1][11b3b26d][ExtensionHostConnection] New connection established.
[11010 ms] [18:57:03] [::ffff:127.0.0.1][11b3b26d][ExtensionHostConnection] <222> Launched Extension Host Process.
[11096 ms] [18:57:03] [::ffff:127.0.0.1][aa1b9ff3][ManagementConnection] New connection established.
[11178 ms] [18:57:03] Listing 0 persistent terminals, 0 total terminals
```

**Comment by sbryfcz** [comment]
I meant to attach that image. In the extensions viewlet, I don't see any installed extensions and things like the python debugger don't work. I am able to install it through the VS GUI, but it won't persist upon restart (which I wouldn't expect). I did have a coworker successfully run on a windows machine. Trying to figure out what might be wrongly configured with my mac.

<img width="398" alt="Screen Shot 2021-01-06 at 8 36 14 PM" src="https://user-images.githubusercontent.com/3409729/103840081-d5db5080-505e-11eb-9ae4-0ee26fd68e81.png">

**Comment by Chuxel** [comment]
@chrmarti Any idea what might be happening?

**Comment by chrmarti** [comment]
The log shows `--install-extension ms-python.python` being passed to the server, so it should be installed but there are no log entries from the server beyond `Installing extensions...`.

@sbryfcz Are you behind a network proxy?

@sandy081 Any way to debug this?

**Comment by sbryfcz** [comment]
@chrmarti - I do not believe so. It is a work computer but we have not had to deal with the typical proxy configuration on any other system. I can typically download, pip install, conda install, npm install, etc anything with 0 proxy settings/changes. Using the gui to download the extensions is perfect so I assume if I needed proxy config, those would have failed as well. Maybe I'm mistaken tho.

**Comment by sbryfcz** [comment]
I was playing around and opened a docker session and tried to manually run the `server.sh` command. I got this message (which is different from how the remote container starts). Do you think this is part of the issue? Sorry I'm not more knowledgeable.

```
/home/vscode/.vscode-server/bin/ea3859d4ba2f3e577a159bc91e3074c5d85c0523/server.sh --force-disable-user-env --use-host-proxy --port 0 --extensions-download-dir /home/vscode/.vscode-server/extensionsCache --install-extension ms-python.python --start-server


*
* Visual Studio Code Server
*
* Reminder: You may only use this software with Visual Studio family products,
* as described in the license https://aka.ms/vscode-remote/license
*

Extension host agent listening on 45965

[14:59:24] Extension host agent started.
Installing extensions...
[14:59:24] Error: self signed certificate in certificate chain
    at TLSSocket.onConnectSecure (_tls_wrap.js:1321:34)
    at TLSSocket.emit (events.js:223:5)
    at TLSSocket._finishInit (_tls_wrap.js:794:8)
    at TLSWrap.ssl.onhandshakedone (_tls_wrap.js:608:12) {
  code: 'SELF_SIGNED_CERT_IN_CHAIN'
}
```

**Comment by sandy081** [comment]
Seems the user has proxy set up and extensions installation is failing because of that.

**Comment by sbryfcz** [comment]
I dont _believe_ that is the issue @sandy081, but would be happy to verify somehow. Suggestions? 

I've looked at my VS Code global settings for proxy configuration. With what I believe to be the default settings, I have no problem downloading extensions through the UI. I validated that http_proxy and https_proxy are not set to anything.

<img width="744" alt="Screen Shot 2021-01-07 at 12 49 52 PM" src="https://user-images.githubusercontent.com/3409729/103926463-f4852a00-50e6-11eb-9359-1d1ecbec6a29.png">
